org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/AbstractJavaModelCompletionTests.java

```diff
@@ -217,4 +217,27 @@ protected void assertResults(String expected, String actual) {
 		throw c;
 	}
 }
+
+protected void assertContains(String message, String expected, String containsIn) {
+	if (!containsIn.contains(expected)) {
+		StringBuffer formatted;
+		if (message != null) {
+			formatted = new StringBuffer(message).append('.');
+		} else {
+			formatted = new StringBuffer();
```

Comment:
If this was copy-pasted, is it possible to re-use the original code?

If not: Please use a `StringBuilder` and move the creation to line 223 (where `formatted` is declared). Then we don't need the `else` clause.

Reply:
Its a copy from org.eclipse.jdt.core.tests.junit.extension.TestCase.assertStringEquals which checks for equality. Since its in a different module i didn't want to refactor it.